src/openrct2/object/ObjectJsonHelpers.cpp

```diff
@@ -56,6 +56,10 @@ namespace ObjectJsonHelpers
             auto length = g1_calculate_data_size(&orig);
             g1 = orig;
             g1.offset = (uint8_t*)std::malloc(length);
+            if (g1.offset == nullptr)
```

Comment:
Replace with `new[]` instead.

Reply:
While I see the point of this check, it could really only happen on some odd, maybe hacked, objects, that nobody actually wanting to play the game would use, right? Do we have access to and can we inspect such objects? Is there any other place they could be filtered out?

Reply:
What check are you referring to?

Reply:
the check you added suggests that `length` might be some large value and cause allocation to fail. What objects would have such large value?

Reply:
They probably would not but that that doesn't mean I shouldn't prevent any possible segfault.